packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts

```diff
@@ -174,6 +192,21 @@ export class DbSyncStakePoolProvider extends DbSyncProvider(RunnableModule) impl
 
   public async queryStakePools(options: QueryStakePoolsArgs): Promise<Paginated<Cardano.StakePool>> {
     const { filters, pagination, rewardsHistoryLimit = REWARDS_HISTORY_LIMIT_DEFAULT } = options;
+    const useBlockfrost = this.#useBlockfrost ? options.useBlockfrost !== false : false;
+
+    if (options.useBlockfrost !== undefined && (!this.#useBlockfrost || !this.#enableBlockfrost)) {
+      throw new ProviderError(
+        ProviderFailure.BadRequest,
+        undefined,
+        'The parameter "useBlockfrost" is not allowed in this mode'
+      );
+    }
+
+    this.logger.error(useBlockfrost);
+    this.logger.error(useBlockfrost);
+    this.logger.error(useBlockfrost);
+    this.logger.error(useBlockfrost);
+    this.logger.error(useBlockfrost);
```

Comment:
:smile: ?

Reply:
You discovered one of my secret development techniques! 🤣